CompositionRoot/A.swift

```diff
@@ -4,15 +4,20 @@ protocol AProtocol {
 
 class A: AProtocol {
     
-    let applicationContext: ApplicationContext
-    init(applicationContext: ApplicationContext) {
-        self.applicationContext = applicationContext
+    let f: FProtocol
+    let g: GProtocol
+    let bFactory: BFactoryProtocol
+
+    init(f: FProtocol, g: GProtocol, bFactory: BFactoryProtocol) {
+        self.f = f
+        self.g = g
+        self.bFactory = bFactory
     }
     
     func aMethod() {
-        let b = B(applicationContext: applicationContext)
+        let b = bFactory.createB()
```

Comment:
This line of code makes A does not depend on the specific class `B` and make it depends on abstract protocol `BFactoryProtocol`.  That's very nice.